File: practical4.ts
function findPeakElement(nums: number[]): number {

  var r = nums.length - 1	// right
  var l = 0			// left


  while(l < r) {
    var mid = Math.floor(l + (r - l) / 2)

    if (nums[mid] < nums[mid+1]) {
      l = mid + 1
    } 

    else {
      r = mid
    }
  }

  return l
};


const nums = [1, 2, 3, 1]
console.log(findPeakElement(nums))	// output: 2

// you may have to reset the nums variable before this works
// const nums = [3]
// console.log(FPE(nums))	// output: 0

File: practical4.py
class Solution:
    def findPeakElement(self, nums: List[int]) -> int:
        left, right = 0, len(nums) - 1
        while left < right:
            mid = (left + right)
            if nums[mid] > nums[mid + 1]:  
                right = mid  			# Search in the left half (including mid)
	    else:  				# Executes only if mid is not a peak
                left = mid + 1
        return left  				# left will always be the peak element


nums = [1, 2, 3, 1]
print(findPeakElement(nums)	# output: 2

print(findPeakElement([3])	# output: 0
